Fix 711 in FIX

foroughgoudarzi opened the pull request.

Commit:
Fix #711

JerryShea commented on src/main/java/net/openhft/chronicle/network/AlwaysStartOnPrimaryConnectionStrategy.java:
```diff
@@ -144,6 +147,24 @@ public long pauseMillisBeforeReconnect() {
+    public AlwaysStartOnPrimaryConnectionStrategy minTimeSec(long minTimeSec) {
+        this.minTimeSec = minTimeSec;
+        return this;
+        }
```
formatting

JerryShea commented on src/main/java/net/openhft/chronicle/network/ConnectionStrategy.java:
```diff
@@ -183,5 +183,12 @@ default void throwExceptionIfClosed() throws IllegalStateException {
     }
-    
+
+     long minTimeSec();
```
I don't believe it is necessary to add these to `ConnectionStrategy`

JerryShea commented on src/main/java/net/openhft/chronicle/network/AlwaysStartOnPrimaryConnectionStrategy.java:
```diff
@@ -51,6 +51,9 @@
     private int socketConnectionTimeoutMs = Integer.getInteger("connectionStrategy.socketConnectionTimeoutMs", 1);
     private long pauseMillisBeforeReconnect = Integer.getInteger("connectionStrategy.pauseMillisBeforeReconnect", 500);
     private ClientConnectionMonitor clientConnectionMonitor = new VanillaClientConnectionMonitor();
+    private long minTimeSec = Integer.getInteger("connectionStrategy.pause.min.secs", 5);
```
I think you only need to change this file?

JerryShea replied:
ah or else maybe just default get methods in `ConnectionStrategy` like we do for `pauseMillisBeforeReconnect`

alamar replied:
Please always use Jvm.getInteger() because it allows code user to set those properties in advance in controlled fashion.

foroughgoudarzi replied:
Thanks, already did, see the last changes.

Commit:
update

JanStureNielsen commented on src/main/java/net/openhft/chronicle/network/AlwaysStartOnPrimaryConnectionStrategy.java:
```diff
@@ -51,6 +51,9 @@
     private int socketConnectionTimeoutMs = Integer.getInteger("connectionStrategy.socketConnectionTimeoutMs", 1);
     private long pauseMillisBeforeReconnect = Integer.getInteger("connectionStrategy.pauseMillisBeforeReconnect", 500);
     private ClientConnectionMonitor clientConnectionMonitor = new VanillaClientConnectionMonitor();
+    private long minTimeSec = Integer.getInteger("connectionStrategy.pause.min.secs", 5);
```
This property name introduces yet another naming scheme to the configuration properties; consider using one of the existing, e.g., `connectionStrategy.minimumPauseSeconds` and `connectionStrategy.maximumPauseSeconds`?

And perhaps open a new issue to address the `Ms` vs `Millis` and lack of units in the other property names.

JerryShea replied:
@JanStureNielsen these property names are legacy, and in use by customers

JanStureNielsen replied:
Oh, bummer -- OT: is there support for alias so the code-base can be consistently named while retaining the legacy configuration name and presenting a more consistent naming scheme?

RobAustin replied:
> This property name introduces yet another naming scheme to the configuration properties; consider using one of the existing, e.g., `connectionStrategy.minimumPauseSeconds` and `connectionStrategy.maximumPauseSeconds`?
> 
> And perhaps open a new issue to address the `Ms` vs `Millis` and lack of units in the other property names.

I agree - these new suggested method names are better

JerryShea replied:
> > This property name introduces yet another naming scheme to the configuration properties; consider using one of the existing, e.g., `connectionStrategy.minimumPauseSeconds` and `connectionStrategy.maximumPauseSeconds`?
> > And perhaps open a new issue to address the `Ms` vs `Millis` and lack of units in the other property names.
> 
> I agree - these new suggested method names are better

Yes but that is changing property names that customers are using. If we change then we need to support the old for a period. If we are going to change we should review all the system properties and consistentise them

JanStureNielsen commented on src/main/java/net/openhft/chronicle/network/AlwaysStartOnPrimaryConnectionStrategy.java:
```diff
@@ -144,6 +147,24 @@ public long pauseMillisBeforeReconnect() {
+        return minTimeSec;
+    }
+
+    public AlwaysStartOnPrimaryConnectionStrategy minTimeSec(long minTimeSec) {
```
Perhaps more descriptively: `minimumPauseSeconds`?

JanStureNielsen commented on src/main/java/net/openhft/chronicle/network/AlwaysStartOnPrimaryConnectionStrategy.java:
```diff
@@ -144,6 +147,24 @@ public long pauseMillisBeforeReconnect() {
         return pauseMillisBeforeReconnect;
     }
 
+    public long minTimeSec() {
```
`minimumPauseSeconds`?

JanStureNielsen commented on src/main/java/net/openhft/chronicle/network/AlwaysStartOnPrimaryConnectionStrategy.java:
```diff
@@ -144,6 +147,24 @@ public long pauseMillisBeforeReconnect() {
+        return this;
+        }
+
+    public AlwaysStartOnPrimaryConnectionStrategy maxTimeSec(long maxTimeSec) {
```
`maximumPauseSeconds`?

JanStureNielsen commented on src/main/java/net/openhft/chronicle/network/AlwaysStartOnPrimaryConnectionStrategy.java:
```diff
@@ -144,6 +147,24 @@ public long pauseMillisBeforeReconnect() {
+        return this;
+        }
+
+    public long maxTimeSec() {
```
`maximumPauseSeconds`?

JerryShea replied:
Yeah agree that we should not call them maxTimeSec - max time of what? Should be maxPauseSec IMO

JerryShea commented on src/main/java/net/openhft/chronicle/network/ConnectionStrategy.java:
```diff
@@ -183,5 +183,12 @@ default void throwExceptionIfClosed() throws IllegalStateException {
-    
+
+    default long minTimeSec() {
+        return 5;
```
so I think these are the only methods that need to be added, and they should refer to the properties e.g. `Integer.getInteger("connectionStrategy.pause.min.secs", 5)`

I don't believe we need any changes to AlwaysStart... and FatalFailre...

JerryShea replied:
These should refer to the system properties. AlwaysStartOnPrimaryConnectionStrategy can refer to these thus ensuring system properties are in one place

JerryShea replied:
@foroughgoudarzi please see my last comment

foroughgoudarzi replied:
All system properties are in one place (AlwaysStartOnPrimaryConnectionStrategy). Please let me know where would you like to define these two system properties? ConnectionStrategy is an interface and its variable would be final and not configurable. I have added these system properties the same way that pauseMillisBeforeReconnect has been defined.

JerryShea replied:
We have to have the system property in the interface so that all implementations (even customer impls) use it, and we don't get behaviour changes
```
default long minTimeSec() {
        return Integer.getInteger("connectionStrategy.pause.min.secs", 5);
}
```
and in AlwaysStart... you can refer to the interface to default the field so we minimise duplication
```
private long minPauseSec = ConnectionStrategy.super.minTimeSec();
```
Don't forget to squash merge all these commits!

Commit:
update

foroughgoudarzi commented:
updated the variable names and converted the changes of AlwaysStart... and FatalFailre... and moved the system properties to the ConnectionStrategy interface. 
I inferred from the ticket title that we want to configure these variables from the YAML so I defined the system properties in AlwaysStart... and FatalFailre...  first.

Commit:
Merge branch 'develop' into issue711

JerryShea commented on src/main/java/net/openhft/chronicle/network/AlwaysStartOnPrimaryConnectionStrategy.java:
```diff
@@ -45,11 +45,11 @@
-    private int socketConnectionTimeoutMs = Jvm.getInteger("connectionStrategy.socketConnectionTimeoutMs", 1);
-    private long pauseMillisBeforeReconnect = Jvm.getInteger("connectionStrategy.pauseMillisBeforeReconnect", 500);
+    private final transient AtomicBoolean isClosed = new AtomicBoolean(false);
+    private int tcpBufferSize = Integer.getInteger("tcp.client.buffer.size", TCP_BUFFER);
```
why change from Jvm.getInteger to Integer.getInteger?

JerryShea commented:
> updated the variable names and converted the changes of AlwaysStart... and FatalFailre... and moved the system properties to the ConnectionStrategy interface. I inferred from the ticket title that we want to configure these variables from the YAML so I defined the system properties in AlwaysStart... and FatalFailre... first.

Ah yes.... sorry... You have a good point there. We will need the fields defined in `AlwaysStart...` so we can set with YAML

Commit:
Moved System properties to AlwaysStartOnPrimaryConnectionStrategy
Changed Integer.getInteger to Jvm.getInteger

minborg commented:
Generally, variables involving time are named `fooS`, `barMs` and `bazzNs` using the SI time units with capital letters. Plural forms should be avoided.

JerryShea commented on src/main/java/net/openhft/chronicle/network/ConnectionStrategy.java:
```diff
@@ -40,6 +40,7 @@
 
 public interface ConnectionStrategy extends Marshallable, java.io.Closeable {
 
+
```
added unnecessary change

JerryShea commented on src/main/java/net/openhft/chronicle/network/connection/FatalFailureConnectionStrategy.java:
```diff
@@ -202,4 +202,5 @@ public void readMarshallable(@NotNull WireIn wire) throws IORuntimeException {
         super.readMarshallable(wire);
         init();
     }
+
```
don't need this extra line

Commit:
removed accidentally added spaces

Commit:
update
